File: CountryCount/csharp/UnitTests/UnitTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using CountryCount;

namespace UnitTests
{
    [TestClass]
    public class UnitTests
    {
        [TestMethod]
        public void Test_Solution_ManyCountries()
        {
            var A = new int[7][];

            A[0] = new[] { 5, 4, 4 };
            A[1] = new[] { 4, 3, 4 };
            A[2] = new[] { 3, 2, 4 };
            A[3] = new[] { 2, 2, 2 };
            A[4] = new[] { 3, 3, 4 };
            A[5] = new[] { 1, 4, 4 };
            A[6] = new[] { 4, 1, 1 };

            var c = new Solution();
            int result = c.solution(A);
            Assert.AreEqual(11, result);
        }
    }
}

File: CountryCount/java/src/tests/UnitTests.java
package tests;

import static org.junit.Assert.*;

import org.junit.Test;

import countrycount.Solution;

public class UnitTests {

	@Test
	public void testSolution_CanCountCountries() {
		int[][] A = new int[7][3];
		
        A[0][0] = 5;    A[0][1] = 4;    A[0][2] = 4;
        A[1][0] = 4;    A[1][1] = 3;    A[1][2] = 4;
        A[2][0] = 3;    A[2][1] = 2;    A[2][2] = 4;
        A[3][0] = 2;    A[3][1] = 2;    A[3][2] = 2;
        A[4][0] = 3;    A[4][1] = 3;    A[4][2] = 4;
        A[5][0] = 1;    A[5][1] = 4;    A[5][2] = 4;
        A[6][0] = 4;    A[6][1] = 1;    A[6][2] = 1;
        
        int result = Solution.solution(A);
        
        assertEquals(11, result);
	}

}
